worker/src/unstract/worker/worker_interface.py

```diff
@@ -0,0 +1,77 @@
+from abc import ABC, abstractmethod
+
+
+class UnstratcWorkerInterface(ABC):
+    @abstractmethod
+    def __init__(self, image_name: str, image_tag: str):
+        pass
+
+    @abstractmethod
+    def _get_image(self) -> str:
+        pass
+
+    @abstractmethod
+    def _image_exists(self, image_name_with_tag: str) -> bool:
+        pass
+
+    @abstractmethod
+    def normalize_container_name(self, name: str) -> str:
+        pass
```

Comment:
This will be common for both k8s and docker I believe. So instead of making it an abstract class we could implement it here itself.